src/entitysdk/client.py

```diff
@@ -368,22 +372,33 @@ def download_file(
             project_context: Optional project context.
             token: Authorization access token.
         """
-        url = (
-            route.get_assets_endpoint(
-                api_url=self.api_url,
-                entity_type=entity_type,
-                entity_id=entity_id,
-                asset_id=asset_id,
-            )
-            + "/download"
+        asset_endpoint = route.get_assets_endpoint(
+            api_url=self.api_url,
+            entity_type=entity_type,
+            entity_id=entity_id,
+            asset_id=asset_id,
         )
         token = self._get_token(override_token=token)
         context = self._optional_user_context(override_context=project_context)
+        asset = core.get_entity(
+            asset_endpoint,
+            entity_type=Asset,
+            token=token,
+            project_context=context,
+            http_client=self._http_client,
+        )
+        path: Path = Path(output_path)
+        path = (
+            path / asset.path
+            if path.is_dir()
+            else validate_filename_extension_consistency(path, Path(asset.path).suffix)
```

Comment:
The docstring can be made clearer by saying explicitly that output_path can be a file or a directory that already exists. 
Alternatively, we could have an additional parameter to tell if it's a directory (and create it if it doesn't exist)